src/xenia/kernel/xboxkrnl/xboxkrnl_crypt.cc

```diff
@@ -660,6 +660,27 @@ dword_result_t XeKeysObscureKey(lpvoid_t input, lpvoid_t output) {
 }
 DECLARE_XBOXKRNL_EXPORT1(XeKeysObscureKey, kNone, kImplemented);
 
+dword_result_t XeKeysHmacShaUsingKey(lpvoid_t obscured_key, lpvoid_t inp_1,
+                                     dword_t inp_1_size, lpvoid_t inp_2,
+                                     dword_t inp_2_size, lpvoid_t inp_3,
+                                     dword_t inp_3_size, lpvoid_t out,
+                                     dword_t out_size) {
+  if (obscured_key) {
+    uint8_t key[16];
+
+    // Deobscure key
```

Comment:
Since this may be needed in other XeKeys functions if there are any, would be nice to move this to a `static` or `inline` function, but not necessary for this pull request.